packages/flame/test/components/component_test.dart

```diff
@@ -1022,16 +1023,26 @@ void main() {
 
     group('Rebalancing components', () {
       testWithFlameGame(
-        'rebalance is queued',
+        'rebalance is correctly queued',
         (game) async {
           final c = Component();
           await game.world.add(c);
 
           c.priority = 10;
-          expect(c.priority, 0);
+          expect(c.priority, 10);
+          expect(
+            game.queue.any(
+              (e) =>
+                  e.child == c &&
+                  e.parent == game.world &&
+                  e.kind == LifecycleEventKind.rebalance,
+            ),
+            isTrue,
+          );
```

Comment:
can we also check the order on the set is untouched? (can be followup!)

Reply:
actually I'd rather only check that than to make everything public

Reply:
Don't we already do that in other tests? :thinking:

Reply:
Maybe we don't, I'll create a test for that!